### DISCUSSION TITLE
Enabling 64k page size for Oracle Cloud Ampere A1

### DISCUSSION THREAD

**Discussion by martinmanyhats** [ptitSeb/box64#339] [Announcements]
I have an Oracle Cloud instance which is installed with AlmaLinux 8.6 (aka CentOS 8.6), which has a 64k page size. This page size is not currently supported by box64.

Without this option a build with:

$ cmake .. -DRPI4ARM64=1 -DCMAKE_BUILD_TYPE=RelWithDebInfo

When running a simple ELF 64-bit LSB executable (classic hello world in C):

./box64 ../../hello/hello
Dynarec for ARM64, with extension: ASIMD AES CRC32 PMULL ATOMICS PageSize:65536
Box64 with Dynarec v0.1.9 452a390 built on Jun 30 2022 11:07:01
Error: PageSize configuation is wrong: configured with 4096, but got 65536

I have created a PR which allows for building box86 with a 64k page size:

https://github.com/ptitSeb/box64/pull/334

Resulting in:

./box64 ../../hello/hello
Dynarec for ARM64, with extension: ASIMD AES CRC32 PMULL ATOMICS PageSize:65536
Box64 with Dynarec v0.1.9 826f9cf built on Jun 30 2022 10:09:17
Using default BOX64_LD_LIBRARY_PATH: ./:lib/:lib64/:x86_64/:bin64/:libs64/
Using default BOX64_PATH: ./:bin/
Counted 21 Env var
Looking for ../../hello/hello
Rename process to "hello"
Using native(wrapped) libc.so.6
Using native(wrapped) ld-linux-x86-64.so.2
Using native(wrapped) libpthread.so.0
Using native(wrapped) librt.so.1
hello world

Hope this helps anyone in a similar situation.

martin

**Comment by ptitSeb** [comment]
I converted the issue to a discussion, as you already solved it.

**Comment by martinmanyhats** [comment]
Thanks. My first FOSS submission, wasn't entirely sure of the protocol :)

martin

**Comment by ptitSeb** [comment]
No problems :)

And thank you for your contribution.
